src/temp_sensor.rs

```diff
@@ -0,0 +1,149 @@
+#[cfg(not(any(esp32)))]
+use esp_idf_sys::{
+    esp, soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_DEFAULT,
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST,
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL,
+    temperature_sensor_clk_src_t, temperature_sensor_config_t, temperature_sensor_disable,
+    temperature_sensor_enable, temperature_sensor_get_celsius, temperature_sensor_handle_t,
+    temperature_sensor_install, temperature_sensor_uninstall, EspError,
+};
+
+// -- TemperatureSensorClockSource --
+
+#[derive(Copy, Clone)]
+#[cfg(not(any(esp32)))]
```

Comment:
Here, and everywhere else further below: why is the esp32 MCU not supported?
Also clippy will bite you for `not(any(esp32))`. Should be just `not(esp32)`